### PR TITLE
feat: RESET_VALUE

### LINKED ISSUE #34: Add support for new RESET_VALUE sentinel

### LINKED ISSUE #35: All unspecified function configuration should be set to NULL. On function deploy, the unspecified option will be reverted back to platform default.
See https://github.com/firebase/firebase-functions/blob/6d4c0a5e69a0df57d64cb095be29facf645b1293/src/runtime/manifest.ts#L166-L177 for JS impl.

### PR DESCRIPTION
**Pull request by Salakar**
Closes #34 
Closes #35

### CODE REVIEW

File: src/firebase_functions/options.py [hunk 1]
```diff
@@ -222,7 +234,27 @@ def _asdict_with_global_options(self) -> dict:
...
+            "service_account",
+            "vpc_connector",
+            "vpc_connector_egress_settings",
+        ]
+        if not preserve_external_changes:
+            for option in resettable_options:
+                if option not in merged_options:
+                    merged_options[option] = RESET_VALUE
+        if "preserve_external_changes" in merged_options:
+            del merged_options["preserve_external_changes"]
```
Comment (taeold) [inline src/firebase_functions/options.py:257]: q: why is this logic necessary

File: src/firebase_functions/private/manifest.py [hunk 2]
```diff
@@ -87,7 +87,7 @@ class BlockingTrigger(_typing.TypedDict):
 
 class VpcSettings(_typing.TypedDict):
     connector: _typing_extensions.Required[str]
-    egressSettings: _typing_extensions.NotRequired[str]
+    egressSettings: _typing_extensions.NotRequired[str | _util.Sentinel]
```
Comment (taeold) [inline src/firebase_functions/private/manifest.py:90]: do you know why we don't need to support similar type fix (i.e. adding _util.Sentinel) to every other option that's resettable?

**Review by taeold — APPROVED** [review]

File: src/firebase_functions/private/manifest.py (hunk 2 above)
Comment (Salakar) [reply src/firebase_functions/private/manifest.py:90]: Other options that were resettable already had this from the original DEFAULT_VALUE - this one just got missed originally

File: src/firebase_functions/options.py (hunk 1 above)
Comment (Salakar) [reply src/firebase_functions/options.py]: It's not needed, have pushed a change to remove it now

**Pull request merged**